src/Areas/Monitor/Services/MonitorMetricsService.cs

```diff
@@ -0,0 +1,334 @@
+                };
+
+                foreach (var timeSeries in metric.TimeSeries)
+                {
+                    if (timeSeries.Values.Count == 0)
+                        continue;
+
+                    var compactTimeSeries = new MetricTimeSeries
+                    {
+                        Metadata = new Dictionary<string, string>(),
+                        Start = timeSeries.Values.First().TimeStamp.UtcDateTime,
+                        End = timeSeries.Values.Last().TimeStamp.UtcDateTime,
+                        Interval = interval ?? "PT1M"
+                    };
+
+                    // Add metadata/dimensions
+                    if (timeSeries.Metadata != null)
+                    {
+                        foreach (var kvp in timeSeries.Metadata)
+                        {
+                            compactTimeSeries.Metadata[kvp.Key] = kvp.Value?.ToString() ?? string.Empty;
+                        }
+                    }
+
+                    // Extract values into arrays, only including non-null arrays
+                    var avgValues = timeSeries.Values
+                        .Where(v => v.Average.HasValue)
+                        .Select(v => v.Average!.Value)
+                        .ToArray();
+                    if (avgValues.Length > 0)
+                        compactTimeSeries.AvgBuckets = avgValues;
+
+                    var minValues = timeSeries.Values
+                        .Where(v => v.Minimum.HasValue)
+                        .Select(v => v.Minimum!.Value)
+                        .ToArray();
+                    if (minValues.Length > 0)
+                        compactTimeSeries.MinBuckets = minValues;
+
+                    var maxValues = timeSeries.Values
+                        .Where(v => v.Maximum.HasValue)
+                        .Select(v => v.Maximum!.Value)
+                        .ToArray();
+                    if (maxValues.Length > 0)
+                        compactTimeSeries.MaxBuckets = maxValues;
+
+                    var totalValues = timeSeries.Values
+                        .Where(v => v.Total.HasValue)
+                        .Select(v => v.Total!.Value)
+                        .ToArray();
+                    if (totalValues.Length > 0)
+                        compactTimeSeries.TotalBuckets = totalValues;
+
+                    var countValues = timeSeries.Values
+                        .Where(v => v.Count.HasValue)
+                        .Select(v => v.Count!.Value)
+                        .ToArray();
+                    if (countValues.Length > 0)
+                        compactTimeSeries.CountBuckets = countValues;
+
+                    compactResult.TimeSeries.Add(compactTimeSeries);
+                }
+
+                results.Add(compactResult);
+            }
+
+            return results;
+        }
+        catch (Exception ex)
+        {
+            throw new Exception($"Error querying metrics: {ex.Message}", ex);
+        }
```

Comment:
Consider just removing the try/catch if we are not going to be doing anything useful besides throwing the same message back up through the caller.

Reply:
Same for all try/catch where we are not recovery or throwing any new more specific exception types.